src/components/Card.js

```diff
@@ -1,61 +1,135 @@
 export default class Card {
-  constructor(cardData, cardSelector, handleCardClick) {
+  constructor(
+    {
+      cardData,
+      cardSelector,
+      handlePreviewImage,
+      userId,
+      handleLikeButton,
+      handleDeleteButton,
+    },
+    myId
+  ) {
     this._name = cardData.name;
     this._link = cardData.link;
+    this._likes = cardData.likes;
+    this._ownerId = cardData.owner._id;
+    this._myId = myId;
+    this._id = cardData._id;
+    this._userId = userId; // not using this value as of now
     this._cardSelector = cardSelector;
-    this._handleCardClick = handleCardClick;
+    this._handlePreviewImage = handlePreviewImage;
+    this._handleDeleteButton = handleDeleteButton;
+    this._handleLikeButton = handleLikeButton;
   }
 
-  _getTemplate() {
-    const cardElement = document
-      .querySelector(this._cardSelector)
-      .content.querySelector(".card")
-      .cloneNode(true);
+  _handleDeleteButtonClick() {
+    this._handleDeleteButton(this._id);
+  }
 
-    return cardElement;
+  getId() {
+    return this._id;
+  }
+
+  isLiked() {
+    return this._likes.some((like) => like._id === this._myId);
+  }
+
+  setLikes(likes) {
+    console.log(likes);
+    this._likes = likes;
+    this._renderLikes();
+  }
+
+  _renderLikes() {
+    this._cardLikesCountElement =
+      this._cardElement.querySelector(".card__likes-count");
+    this._cardLikesCountElement.textContent = this._likes.length;
+
+    const cardLikeButton =
+      this._cardElement.querySelector(".card__like-button");
+    if (this.isLiked()) {
+      cardLikeButton.classList.add("card__like-button_active");
+    } else {
+      cardLikeButton.classList.remove("card__like-button_active");
+    }
+  }
+
+  _fillCardTemplate() {
+    const imageElement = this._cardElement.querySelector(".card__image");
+    imageElement.src = this._link;
+    imageElement.alt = this._name;
+    this._cardElement.id = this._id;
+
+    this._cardElement.querySelector(".card__title").textContent = this._name;
+    this._cardLikesCountElement =
+      this._cardElement.querySelector(".card__likes-count");
+
+    if (this._ownerId !== this._myId) {
+      this._cardElement.querySelector(".card__delete-button").remove();
+    }
+  }
```

Comment:
`  _fillCardTemplate() {
` 

You don't use this method. Please, delete it to clean up the code

Reply:
Removed.